### PR TITLE
[T-61]: EmailAPI

### PR DESCRIPTION
**Pull request by michalpulpan**
This PR implements `EmailAPI` using Django SMTP library. 
As of now, there're 3 types of emails:
* Order confirmation (should be called somewhere in #284 )
* Order review (should be called somewhere in #215 )
* Return confirmation (should be called somewhere in #236 )

This PR also added `redis` into our `docker-compose` for Redis queue and `rq-worker` which is an instance of `backend` serving as worker of Redis queue.

In order to use `Redis Queue`, you have to add to your `docker-compose.env`:
```env
USING_REDIS_QUEUE=1
REDIS_QUEUE_LOCATION=redis
```

And I created test e-mail inbox, so just add into your `docker-compose.env`:
```env
EMAIL_USE_SSL=1
EMAIL_PORT=465
EMAIL_HOST=smtp.seznam.cz
EMAIL_HOST_USER=ecoseller-storefront@email.cz
EMAIL_HOST_PASSWORD=SEE_SLACK 😉 
EMAIL_FROM=Storefront<ecoseller-storefront@email.cz>
```

Hope you like it!

## Update 13. 6.

I have finished email templates and tested/integrated NotificationAPI (see it in the code :) )
So now after me merge order submission and this, we should have emails sent after order submission via our notifiaction API 💯 💯 💯 💯 💯 💯 
Snímek obrazovky 2023-06-13 v 13 40 01

(it's in dark mode, by default it has white background 😊 )

### CODE REVIEW

**Comment by michalpulpan** [comment]
btw. 2k LOC are probably due to HTML files I added (email tempaltes)... see how it works with Django templating engine, it's quite nice and easy.
+ we can extend it very well.

**Review by lakatop — APPROVED** [review]
Damn, you're an implementation machine!

File: src/backend/docker-compose.env
Comment (VL-CZ) [inline src/backend/docker-compose.env:1]: we should IMO put this file to gitignore, since it contains password

**Review by VL-CZ — APPROVED** [review]

File: src/backend/docker-compose.env
Comment (michalpulpan) [reply src/backend/docker-compose.env:1]: `docker-compose.env` is now deleted and should be ignored by `src/backend/.gitignore`

**Pull request merged**